docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+
+TODO: check `FullyLabeledReplicas`, do we still need it?
+
+#### MachineSet (New)Conditions
+
+| Condition        | Note                                                                                                           |
+|------------------|----------------------------------------------------------------------------------------------------------------|
+| `ReplicaFailure` | This condition surfaces issues on creating Machines, if any.                                                   |
+| `MachinesReady`  | This condition surfaces detail of issues on the controlled machines, if any.                                   |
+| `ScalingUp`      | True if available replicas < desired replicas                                                                  |
+| `ScalingDown`    | True if replicas > desired replicas                                                                            |
+| `UpToDate`       | True if all the Machines controlled by this MachineSet are up to date (replicas = upToDate replicas)           |
+| `Remediating`    | True if there is at least one Machine controlled by this MachineSet is not passing health checks               |
+| `Deleted`        | True if MachineSet is deleted; Reason can be used to observe the cleanup progress when the resource is deleted |
+| `Paused`         | True if this MachineSet or the Cluster it belongs to are paused                                                |
+
+> To better evaluate proposed changes, below you can find the list of current MachineSet's conditions:
+> Ready, MachinesCreated, Resized, MachinesReady.
+
+Notes:
+- MachineSet conditions are intentionally mostly consistent with MachineDeployment conditions to help users troubleshooting .
+- MachineSet is considered as a sort of implementation detail of MachineDeployments, so it doesn't have its own concept of availability.
+  Similarly, this proposal is dropping the notion of MachineSet readiness because it is preferred to let users focusing on Machines readiness.
+- `Remediating` for older MachineSet sets will report that remediation will happen as part of the regular rollout.
+- `UpToDate` condition initially will be `false` for older MachineSet, `true` for the current MachineSet; however in
+  the future the latter might evolve in case Cluster API will start supporting in-place upgrades.
+
+#### MachineSet Print columns
+
+| Current       | To be                   |
+|---------------|-------------------------|
+| `NAME`        | `NAME`                  |
+| `CLUSTER`     | `CLUSTER`               |
+| `DESIRED` (*) | `PAUSED` (new) (*)      |
+| `REPLICAS`    | `DESIRED`               |
+| `READY`       | `CURRENT` (renamed) (*) |
+| `AVAILABLE`   | `READY` (updated)       |
+| `AGE`         | `AVAILABLE` (updated)   |
+| `VERSION`     | `UP-TO-DATE` (new)      |
+|               | `AGE`                   |
+|               | `VERSION`               |
```

Comment:
Could we maybe also add the name of the bootstrapRef and infraRef resource? or is that too much info?

Reply:
Ideally we can, but I think we should figure it out when we implement it and we can see how it looks like.
Also, print columns are not subject to API guarantee, so we are free to iteratively improve this anytime.

Reply:
Added a not about this, resolving

(also trying to keep the number of open threads under control 😅 )